commands/up.go

```diff
@@ -67,17 +77,88 @@ func upHandler(cmd *cobra.Command, args []string) error {
 	if err := runBuild(cmd, args); err != nil {
 		return err
 	}
-	fmt.Println()
+
 	if !skipPush {
 		if err := runPush(cmd, args); err != nil {
 			return err
 		}
-		fmt.Println()
 	}
+
 	if !skipDeploy {
 		if err := runDeploy(cmd, args); err != nil {
 			return err
 		}
 	}
+
+	if watch {
+		logger := log.New(os.Stderr, "", log.LstdFlags)
+		logger.SetFlags(log.Ldate | log.Ltime | log.LUTC)
+
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			return err
+		}
+		defer watcher.Close()
+
+		err = filepath.Walk("./", func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return watcher.Add(path)
+			}
```

Comment:
I think it makes sense to have a list of non-watched paths, at a minimum, `.git` and the `build` folder should be omitted. 

```go
var skipWatch = map[string]struct{}{
    ".git",
    "build",
}
```

In fact, it is probably reasonable to omit _all_ hidden directories, for example `.vscode`  and `.pytest_cache` are not relevant. There are probably other paths that we should also reasonably ignore, in a  python project `__pycache__` could also be ignored and in a nodejs project, maybe `node_modules`. Perhaps we could read the `.gitignore` file and omit any path that matches one of the patterns. This would allow maximum flexibility and have a high probably of matching files that don't actually indicate a change in the functions. 

Alternatively, the stack file should be read and watches only created for each [`handler`](https://docs.openfaas.com/reference/yaml/#function-handler), but this is not enough, you would  also need to include any paths from the [`copy` field](https://docs.openfaas.com/reference/yaml/#copy), if it exists. 

For simplicity it is probably enough to skip `.git` and any path that matches a pattern in the `.gitignore`

Reply:
@LucasRoesler Made changes for skipping `.git` file and files specified in the `.gitignore` file for the watch. 
PTAL: https://github.com/openfaas/faas-cli/pull/962/commits/5235a62b1b2599ae50ce5355915fe833d1e1a8b8

Reply:
Since `build` is not initially present in the dir, so I think we need to also go through the `filpath.Walk` after first build for skipping the build dir for skipping watch on that. Please give your thoughts @LucasRoesler @alexellis.